client/host_volume_endpoint.go

```diff
@@ -23,31 +22,44 @@ func newHostVolumesEndpoint(c *Client) *HostVolume {
 
 var hostVolumeRequestTimeout = time.Minute
 
-func (v *HostVolume) requestContext() (context.Context, context.CancelFunc) {
-	return context.WithTimeout(context.Background(), hostVolumeRequestTimeout)
-}
+func (v *HostVolume) Create(
+	req *cstructs.ClientHostVolumeCreateRequest,
+	resp *cstructs.ClientHostVolumeCreateResponse) error {
 
-func (v *HostVolume) Create(req *cstructs.ClientHostVolumeCreateRequest, resp *cstructs.ClientHostVolumeCreateResponse) error {
 	defer metrics.MeasureSince([]string{"client", "host_volume", "create"}, time.Now())
-	_, cancelFn := v.requestContext()
+	ctx, cancelFn := v.requestContext()
 	defer cancelFn()
 
-	// TODO(1.10.0): call into Client's host volume manager to create the work here
+	cresp, err := v.c.hostVolumeManager.Create(ctx, req)
+	if err != nil {
+		v.c.logger.Debug("failed to create host volume", "name", req.Name, "error", err)
```

Comment:
oh yeah, I've been glazing over this debug message, because the external plugin caller itself logs an ERROR that stands out more, and then also the http handler (usually server, probably?).  the logs look like this on failure:

> [DEBUG] hostvolumemanager/host_volume_plugin.go:133: client.host_volumes: running host volume plugin: plugin_id=example-host-volume operation=create volume_id=98aff98d-c3a6-4914-3b5e-d7b397e4bf23
> [ERROR] hostvolumemanager/host_volume_plugin.go:163: client.host_volumes: error with plugin: plugin_id=example-host-volume operation=create volume_id=98aff98d-c3a6-4914-3b5e-d7b397e4bf23
>   stdout=
>   | uh oh
> 
>   stderr=
>   | plugin had a bad day
>    err="exit status 2"
> [DEBUG] client/host_volume_endpoint.go:35: client: failed to create host volume: name=test error="error creating volume \"98aff98d-c3a6-4914-3b5e-d7b397e4bf23\" with plugin \"example-host-volume\": exit status 2"
> [ERROR] agent/http.go:752: http: request failed: method=PUT path=/v1/volume/host/create error="HostVolume.Create error: error creating volume \\"98aff98d-c3a6-4914-3b5e-d7b397e4bf23\\" with plugin \\"example-host-volume\\": exit status 2" code=500


I'm inclined to keep the lower-level `host_volume_plugin` log, since it has stdout+stderr, and just drop this RPC debug.

actually all of the debug logs in this file are duplicative of the lower-level ones, so I'll drop all these in favor of those, unless there's some reason to also keep the logs in the RPC handler?

Reply:
eh, I'll keep the RPC logs, too, and upgrade to Error. they explicitly say "failed to (create|delete) host volume", and it is sometimes nice to have logs at different layers when trying to track down bugs.

Reply:
I like to think of it in terms of "unit of work" and have at most one error-level log (or info-level log) per unit of work like a single RPC call handler. Then have inner debug-level messages to help out debugging.

Reply:
sure that tracks - then I'll also up the Debugs at this layer to Info (one log line with each volume creation/deletion seems reasonable to me), and downgrade the current Error within the plugin caller to Debug